model/settings/service.go

```diff
@@ -129,3 +133,67 @@ func (s *SettingsService) StartEmailUpdate(inst *instance.Instance, cmd *UpdateE
 
 	return nil
 }
+
+// ConfirmEmailUpdate is the second step to the email update process.
+//
+// This step consiste to make the email change effectif and relay the change
+// into the cloudery.
+func (s *SettingsService) ConfirmEmailUpdate(inst *instance.Instance, tok string) error {
+	settings, err := s.storage.getInstanceSettings(inst)
+	if err != nil {
+		return fmt.Errorf("failed to fetch the settings: %w", err)
+	}
+
+	pendingEmail, ok := settings.M["pending_email"].(string)
+	if !ok {
+		return ErrNoPendingEmail
+	}
+
+	err = s.token.Validate(inst, token.EmailUpdate, pendingEmail, tok)
+	if err != nil {
+		return fmt.Errorf("failed to validate the token: %w", err)
+	}
+
+	settings.M["email"] = pendingEmail
+	delete(settings.M, "pending_email")
+
+	err = s.storage.setInstanceSettings(inst, settings)
+	if err != nil {
+		return fmt.Errorf("failed to save the settings changes: %w", err)
+	}
+
+	err = s.cloudery.SaveInstance(inst, &cloudery.SaveCmd{
+		Locale:     inst.Locale,
+		Email:      settings.M["email"].(string),
+		PublicName: settings.M["public_name"].(string),
```

Comment:
I would prefer to use a local variable with:

```go
publicName, _ := settings.M["public_name"].(string)
```

If an instance has no public_name, it would avoid a panic.